app/tests/components_tests/test_models.py

```diff
@@ -1466,3 +1466,133 @@ def test_displacement_field_validation(
+        (
+            {
+                "name": "test_name",
+                "type": "bounding box",
+                "corners": [[0, 0, 0], [10, 0, 0], [10, 10, 0], [0, 10, 0]],
+                "version": {"major": 1, "minor": 0},
+                "probability": 0.2,
+            },
+            "[\"JSON does not fulfill schema: instance 'bounding box' is not one of ['2D bounding box']\"]",
+            pytest.raises(ValidationError),
+        ),
+        (
+            {
+                "name": "test_name",
+                "type": "2D bounding box",
+                "corners": [[0, 0, 0], [10, 0, 0], [10, 10, 0], [0, 10, 0]],
+                "version": {"major": 1, "minor": 0},
+                "probability": 0.2,
+            },
+            None,
+            nullcontext(),
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_ci_example_value(example_value, expected_value, context):
+
+    sc = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "type": "object",
+        "properties": {
+            "version": {
+                "type": "object",
+                "properties": {
+                    "major": {"type": "integer"},
+                    "minor": {"type": "integer"},
+                },
+                "required": ["major", "minor"],
+            },
+            "type": {"type": "string"},
+            "name": {"type": "string"},
+            "corners": {
+                "type": "array",
+                "items": [
+                    {
+                        "type": "array",
+                        "items": [
+                            {"type": "integer"},
+                            {"type": "integer"},
+                            {"type": "integer"},
+                        ],
+                    },
+                    {
+                        "type": "array",
+                        "items": [
+                            {"type": "integer"},
+                            {"type": "integer"},
+                            {"type": "integer"},
+                        ],
+                    },
+                    {
+                        "type": "array",
+                        "items": [
+                            {"type": "integer"},
+                            {"type": "integer"},
+                            {"type": "integer"},
+                        ],
+                    },
+                    {
+                        "type": "array",
+                        "items": [
+                            {"type": "integer"},
+                            {"type": "integer"},
+                            {"type": "integer"},
+                        ],
+                    },
+                ],
+            },
+            "probability": {"type": "number"},
+        },
+        "required": ["version", "type", "name", "corners", "probability"],
+    }
+
+    ci = ComponentInterfaceFactory(
+        kind=InterfaceKindChoices.TWO_D_BOUNDING_BOX,
+        schema=sc,
```

Comment:
As far as I can tell this test doesn't test your functionality at all. You've set the kind to `InterfaceKindChoices.TWO_D_BOUNDING_BOX` then replicating that schema?! The schema should add extra restrictions on top of the underlying schema. In this case it should be for extra restrictions on top of 2d bounding boxes.

Reply:
This test would be much clearer to test this with a Number kind and then limit that number with a custom schema to greater than zero.

Reply:
Done